src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -86,6 +92,11 @@
 
 public class LibraryTab extends Tab {
 
+    /**
+     * Defines the different modes that the BasePanel can operate in
+     */
+    private enum PanelMode { TABLE, TABLE_EDITOR }
```

Comment:
I would favor  MAIN_TABLE and ENTRY_EDITOR and not table editor

Reply:
Reasoning was that with the Entry Editor both are shown, the table and the editor, while with the entry editor turned off, only the table is. But I have no strong feelings about this.

Reply:
I think, there is still `MainTable.java`. Thus: `MAIN_TABLE`, `MAIN_TABLE_AND_ENTRY_EDITOR` (The `AND` was added to be clear about the options)